### PR TITLE
feat: use build time defined Device Type instead of hardcoded

### PR DESCRIPTION
**Pull request by larsewi**

### CODE REVIEW

**Review by lluiscampos — CHANGES_REQUESTED** [review]
I am missing one thing:

:point_right:  To set `CONFIG_MENDER_DEVICE_TYPE` in the project properties. Here we have a golden opportunity to set it per board, using the `boards/...conf` files

> Merge together with [mendersoftware/mender-mcu#85](https://github.com/mendersoftware/mender-mcu/pull/85)

Not really. We first need to merge `mender-mcu`, then port it to the stable branch there, and only then we can merge this PR in `mender-mcu-integration`

File: src/main.c [hunk 1]
```diff
@@ -33,9 +33,6 @@ LOG_MODULE_REGISTER(mender_app, LOG_LEVEL_DBG);
 #include "modules/noop-update-module.h"
 #endif /* CONFIG_MENDER_APP_NOOP_UPDATE_MODULE */
 
-// TODO: Rework with MEN-7547
-#define DEVICE_TYPE   "espressif-esp32"
```
Comment (vpodzime) [inline src/main.c:37]: Let's please have a reasonable default. I think it should be `CONFIG_BOARD_TARGET`.

**Review by vpodzime — APPROVED** [review]
ACK

**Comment by larsewi** [comment]
> I am missing one thing:
>
> 👉 To set CONFIG_MENDER_DEVICE_TYPE in the project properties. Here we have a golden opportunity to set it per board, using the boards/...conf files

Maybe I'm misunderstanding you. But currently `CONFIG_MENDER_DEVICE_TYPE` is set automatically based on Zephyrs `CONFIG_BOARD_TARGET`. Below is contents in `west build -t menuconfig` after a pristine build:

Screenshot 2024-10-02 at 16 44 31

**Review by lluiscampos — APPROVED** [review]
> @lluiscampos maybe I'm misunderstanding you. But currently `CONFIG_MENDER_DEVICE_TYPE` is set automatically based on Zephyrs `CONFIG_BOARD_TARGET`. Below is contents in `west build -t menuconfig` after a pristine build:

Yes, I completely misunderstood the implementation on my first review. Sorry for the noise. It is fine as-is. Just don't merge it before we get the feature merged into stable branch v0.2 in `mender-mcu`.

**Pull request merged**